### PR TITLE
Post settings excerpt

### LINKED ISSUE #5807: Post Settings: New dialog to edit the excerpt
The current excerpt editor is inlined in the Post Settings. It makes the screen complicated and hard to read. To avoid this, we're going to open a dialog with a text area with a hint.

excerpt

### PR DESCRIPTION
**Pull request by oguzkocer**
Fixes #5807. This is first of many post settings PRs. I wanted to start with something isolated, so picked the excerpt. The design is not fully implemented yet, this PR only handles the functionality of the excerpt as well as the excerpt dialog's design.

We might refactor a few things in the code to make them reusable, however I didn't want to over-engineer it. Those refactors will come when we need them in the following PRs, as well as a full design revamp of the post settings page. Feel free to ping me to ask if something's planned for further PRs to make sure I didn't miss anything. Thanks!

The excerpt is at the top of the page to make it easier to test, it'll move back to where it belongs in the further PRs.

To test:
* Go into the post settings page, tap on the excerpt and change it.
* After you change it, tap on it again, to make sure the dialog shows the updated text
* Remove the excerpt and make sure it's set to "Not Set"
* When the excerpt was removed, update the settings and make sure the excerpt on the remote is empty and not "Not Set"

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java [hunk 1]
```diff
@@ -351,7 +360,11 @@ private void setupSuggestionServiceAndAdapter() {
     }
 
     private void initSettingsFields() {
-        mExcerptEditText.setText(mPost.getExcerpt());
+        if (!TextUtils.isEmpty(mPost.getExcerpt())) {
+            mExcerptTextView.setText(mPost.getExcerpt());
+        } else {
+            mExcerptTextView.setText(getString(R.string.not_set));
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: 👍

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -278,14 +278,18 @@
...
     <string name="download">Downloading media</string>
-    <string name="post_settings">Post settings</string>
     <string name="delete_post">Delete post</string>
     <string name="delete_page">Delete page</string>
     <string name="share_url">Share URL</string>
     <string name="posts_fetching">Fetching posts…</string>
     <string name="pages_fetching">Fetching pages…</string>
     <string name="toast_err_post_uploading">Unable to open post while it\'s uploading</string>
 
+    <!--post settings-->
```
Comment (tonyr59h) [inline WordPress/src/main/res/values/strings.xml]: nitpick; there's a `editor post settings` section on line 980.

**Comment by tonyr59h** [comment]
Deleting the excerpt text after setting it doesn't show `Not Set`.

**Comment by tonyr59h** [comment]
I should clarify that leaving the editor and returning it will correctly show `Not Set`.

**Comment by oguzkocer** [comment]
> Deleting the excerpt text after setting it doesn't show `Not Set`.

Fixed in 83322e34bac450e45a2dfccdce3ccee805f5fc1a. I've also introduced `setPostExcerptToNotSet` method to avoid calling `mExcerptTextView.setText(R.string.not_set);` from multiple places.

Thanks for the review @tonyr59h, ready for another round!

**Comment by tonyr59h** [comment]
Works now! :shipit:

**Pull request merged**